File: go/first-digit/first-digit.go
package main

import(
	"fmt"
	"slices"
)

func firstDigit(inputString string)string{
	start, end := 48 , 57
	numbers := make([]int, 0)
    for i := start; i <= end; i++ {
        numbers = append(numbers, i)
    }
	for _, val := range inputString{
		if slices.Contains(numbers, int(val)){
			return string(rune(val))
		} 
	}
	return "0"
}

func main(){
	inputString := "var_1__Int"
	fmt.Printf("The leftmost digit that occurs in %s is: %s \n", inputString, firstDigit(inputString))
}

File: js/first-digit.js
function firstDigit(inputString){

    //wITHOUT REGULAR EXPRESSIONS
    // let numbers = Array.from(Array(10)).map((e, i) => i + 48).map((x) => String.fromCharCode(x))
    // let stringSplit = inputString.split('')
    // for(let i = 0; i < stringSplit.length; i++){
    //     if(numbers.includes(stringSplit[i])){
    //         return stringSplit[i]
    //     }
    // }

    //WITH REGULAR EXPRESSIONS
    return inputString[inputString.search(/\d/)]
}

let inputString = "var_1__Int"
console.log(`The leftmost digit that occurs in ${inputString} is:`, firstDigit(inputString))